utt/signature-processor/include/sigData.hpp

```diff
@@ -0,0 +1,68 @@
+// UTT
+//
+// Copyright (c) 2020-2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the LICENSE
+// file.
+
+#include <map>
+#include <unordered_map>
+#include <mutex>
+#include <functional>
+#include "types.hpp"
+namespace utt {
+
+struct SigJobEntry {
+  uint64_t job_id;
+  uint64_t start_time{0};
+  uint64_t job_timeout_ms_{1000};
+  std::map<uint32_t, libutt::api::types::Signature> partial_sigs;
+};
+
+class SigJobEntriesMap {
```

Comment:
I find the interface of this class a bit confusing.
* We call "registerEntry" and "put" with a SigJobEntry to actually create an entry? Can we create a valid entry if we call just one of those methods?
* We have an option to call "get" or "getEntries". The first returns a non-const reference, the second returns a copy of the internal map. Supposedly this hints that the "jobs" themselves are the entries, but when we call "registerEntry" we actually add a validation callback.

Reply:
fixed